auth/uauth.go

```diff
@@ -48,7 +48,7 @@ var (
 // CreateAuthInfo interface provides necessary data
 type CreateAuthInfo interface {
 	Config() config.Provider
-	Logger() ulog.Log
+	Logger() *zap.Logger
```

Comment:
I started email thread before this PR, and Im curious too. I have been asked this question multiple times that why fx is so tightly integrated with zap.Logger, and can it be used with different logger implementation.